src/main/java/frc/robot/commands/XPercentSpeed.java

```diff
@@ -0,0 +1,22 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import frc.robot.subsystems.Drivetrain;
+
+public class XPercentSpeed extends InstantCommand {
+
+    private final Drivetrain m_subsystem;
+    private final double m_speedMult;   //The coefficient that the speed of the Drivetrain will be multiplied by (between 0 and 1). This number doesn't matter if the function is turning off the drive mode
+
+    public XPercentSpeed(Drivetrain subsystem, double speedMult) {
```

Comment:
I'd say that the command should take a second parameter which states whether it's on or off instead of a toggle. I think it could get confusing if we're calling this from multiple sources, like if someone hits the quarter speed and half speed buttons at the same time and then releases them the behavior wouldn't be what we want.

Reply:
The reason the code looks a little weird is because I wanted it to be able to act identically to the way the QuarterSpeed and HalfSpeed would interact with each other, and I wanted it to be easy to bind it to buttons. As of now, it can be bound to a button in exactly the same way that HalfSpeed or QuarterSpeed would have been (except for the added parameter), and it should work properly, no extra lines needed. Since everything happens in the initialize(), it should always prioritize the button that was pressed last. Technically the method already states on/off and isn't a toggle, but the command XPercentSpeed always toggles it.

Reply:
Good point. We can always create another instant command if we need explicit control or we can just call it directly from the subsystem